test/testfunctions.cpp

```diff
@@ -1511,6 +1512,41 @@ class TestFunctions : public TestFixture {
         ASSERT_EQUALS("[test.cpp:4]: (portability) The 2nd memset() argument '1.0f+i' is a float, its representation is implementation defined.\n", errout.str());
     }
 
+    void memsetInvalid3rdParam() {
+        check("void foo() {\n"
+              "    char s[10];\n"
+              "    memset(s, 0, '*');\n"
+              "}");
+        ASSERT_EQUALS("[test.cpp:3]: (warning) The size argument of memset() is given as a char constant.\n", errout.str());
+
+        check("void foo(char c) {\n"
+              "    char s[10];\n"
+              "    memset(s, 0, c);\n"
+              "}");
+        ASSERT_EQUALS("", errout.str());
+
+        check("void foo(char val, char c) {\n"
+              "    char s[10];\n"
+              "    memset(s, val, c);\n"
+              "}");
+        ASSERT_EQUALS("", errout.str());
+
+        check("void foo(int val, char c) {\n"
```

Comment:
Imho you should check if the type is really `char` and not `u8` or something like that.

Reply:
I could do so, yes. But given the message text "The size argument of memset() is given as a char while the value argument is of a larger type" I do not think this is really necessary. Even if it is uint8_t, it is still suspicious that the value argument is larger than the size argument. And the message is inconclusive, by the way.

Reply:
> Even if it is uint8_t, it is still suspicious that the value argument is larger than the size argument.

that does not seem strange to me. I guess there could be such code: `uint8_t sz = 64;`.

Reply:
And why do you pass that argument as uint8_t while at the same passing the value argument in a larger integer, even though only one byte of it will be used by memset?

Reply:
I don't know. I am just speculating.